### PR TITLE
Add uniques and constants for unit supply

### PR DESCRIPTION
**Pull request by xlenstra**
Also a bit of code readability, though I haven't checked performance impact of that

### CODE REVIEW

File: core/src/com/unciv/models/ruleset/unique/UniqueType.kt [hunk 1]
```diff
@@ -288,6 +288,10 @@ enum class UniqueType(val text: String, vararg targets: UniqueTarget, val flags:
     StartsWithTech("Starts with [tech]", UniqueTarget.Nation),
     ResearchableMultipleTimes("Can be continually researched", UniqueTarget.Global),
 
+    BaseUnitSupply("[amount] Unit Supply", UniqueTarget.Global),
+    UnitSupplyPerPop("[amount] Unit Supply per [amount] population", UniqueTarget.Global),
```
Comment (yairm210) [inline core/src/com/unciv/models/ruleset/unique/UniqueType.kt:292]: This feels like it should have a cityfilter

Comment (xlenstra) [reply core/src/com/unciv/models/ruleset/unique/UniqueType.kt:292]: I was planning so at first too, but removed it after deciding that would complicate the calculation for supply a lot. Basically for each unique I would have to iterate over all cities and add their population only if the city matched the filter (O(n^2)). If you want I can still add it in

Comment (yairm210) [reply core/src/com/unciv/models/ruleset/unique/UniqueType.kt:292]: I think we should add it, it's O(n) per unique which is exactly the complexity of a unique that sums population count anyway :)

Comment (xlenstra) [reply core/src/com/unciv/models/ruleset/unique/UniqueType.kt]: Should be added now

**Pull request merged**